### ISSUE TITLE
Import warning when virtualfolder is not in INSTALLED_APPS

### ISSUE THREAD

**Issue by julen** [translate/pootle#4442]
Since recently we started to get some warnings related to virtual folders, even if we explicitly make no use of them:
```python
RemovedInDjango19Warning: Model class virtualfolder.models.VirtualFolder doesn't declare an explicit app_label and either isn't in an application in INSTALLED_APPS or else was imported before its application was loaded. This will no longer be supported in Django 1.9.
  class VirtualFolder(models.Model):

RemovedInDjango19Warning: Model class virtualfolder.models.VirtualFolderTreeItem doesn't declare an explicit app_label and either isn't in an application in INSTALLED_APPS or else was imported before its application was loaded. This will no longer be supported in Django 1.9.
  class VirtualFolderTreeItem(models.Model, CachedTreeItem):
```

If I'm not mistaken these are reported since `virtualfolder.models` is being imported regardless of `virtualfolder` being in `INSTALLED_APPS`, so Django complains about it in this case.

While adding the suggested `app_label` silences the warning, I'm not 100% sure this is the right way to proceed.

**Comment by unho** [comment]
The right way to proceed is to add `app_label` and conditionally import `virtualfolder` app.

**Comment by phlax** [comment]
> If I'm not mistaken these are reported since virtualfolder.models is being imported regardless of virtualfolder being in INSTALLED_APPS, so Django complains about it in this case.

as @unho has said we need to fix the virtualfolder app to make it compatible

atm when you install Pootle, you install ``virtualfolders`` - the imports should be explicit and at the top of the modules, even if virtualfolders is not in installed_apps and they are not made use of.

**Comment by unho** [comment]
@julen How did you trigger these warnings? Were you running Django 1.9?

**Comment by julen** [comment]
Django 1.8, simply remove `virtualfolders` from `INSTALLED_APPS` in your settings and any view importing such modules will trigger it.

**Comment by phlax** [comment]
@julen @unho i understand better the problem now - we do need better separation of virtualfolders before we can support django 1.9

**Comment by iafan** [comment]
@julen I see these warnings in our production server logs now. Does it mean that `app_label` no longer silences these, or that we don't have it enabled in production?

**Comment by phlax** [comment]
@iafan i think its because you dont have virtualfolders in INSTALLED_APPS, so the `app_label` has no effect.

I have a couple of PRs that are gradually moving all of the virtualfolder code into a plugin. That will fix this problem properly

**Comment by julen** [comment]
Yeah @iafan, I realized these are now showing up (and often) in the logs, adding up a lot of noise and potentially masking other errors unfortunately.

I checked this again and adding an `app_label` to virtual folders doesn't make any difference at all (it wasn't there), so the problem is that a `models` module from an app which is not installed is being imported — the same happened with the commenting code as reported in #4657.

Since we definitely want to avoid our logs being filled up with these warnings as soon as possible, I believe the interim solution will be to conditionally import the virtualfolder modules.
